### PR TITLE
New pupating zombies 

### PR DESCRIPTION
**Pull request by SariusSkelrets**
#### Summary
Content "Add two more pupating zombies and their decoy variants"

#### Purpose of change
There's currently only two variants of the pupating zombie. I want to add more of them

#### Describe the solution
Add two new pupating zombies:

The pupating brute, a slower brute that uses the existing but unused big_raptor_spawn special attack
and
The hive hulk, weaker and slower than the regular hulk but regenerates and the non-decoy version spawns a flesh-raptor once in a while without dying

Also updated the spawnlists

#### Describe alternatives you've considered
Adding a variant of the dissoluted devourer that eats zombies to create flesh-raptors

#### Testing
Tried to spawn them, had many problems, fixed them all, the monsters work as intended now

### CODE REVIEW

File: data/json/monsters/zed-pupating.json [hunk 1]
```diff
@@ -61,6 +61,74 @@
...
+  {
+    "id": "mon_brute_pupa_decoy",
+    "type": "MONSTER",
+    "name": { "str": "pupating brute" },
+    "description": "This muscular human corpse is wrapped in sticky black fibers that cover everything neck down.  Beneath the wrapping there is strange rhythmic movement grotesque to behold.",
+    "copy-from": "mon_zombie_brute",
+    "delete": { "categories": [ "CLASSIC" ] },
+    "speed": 95,
+    "armor_bash": 8,
+    "armor_cut": 6,
```
Comment (Maleclypse) [inline data/json/monsters/zed-pupating.json:73]:
```suggestion

```

File: data/json/monsters/zed-pupating.json [hunk 2]
```diff
@@ -61,6 +61,74 @@
...
+    }
+  },
+  {
+    "id": "mon_hulk_pupa_decoy",
+    "type": "MONSTER",
+    "name": { "str": "hive hulk" },
+    "description": "This gigantic corpse is wrapped in sticky black fibers that cover everything neck down.  Beneath the wrapping there is strange rhythmic movement grotesque to behold.",
+    "copy-from": "mon_zombie_hulk",
+    "delete": { "categories": [ "CLASSIC" ] },
+    "armor_bash": 8,
```
Comment (Maleclypse) [inline data/json/monsters/zed-pupating.json:103]:
```suggestion

```

File: data/json/monsters/zed-pupating.json [hunk 3]
```diff
@@ -61,6 +61,74 @@
...
+    "armor_cut": 6,
+    "armor_bullet": 6,
+    "melee_skill": 4,
+    "melee_dice": 3,
+    "melee_dice_sides": 7,
+    "hp": 300,
+    "speed": 85,
+    "special_attacks": [ { "type": "SMASH", "cooldown": 30 } ],
```
Comment (Maleclypse) [inline data/json/monsters/zed-pupating.json:111]: somehow this line isn't matching acceptable special attacks that exist.

File: data/json/monsters/zed-pupating.json [hunk 4]
```diff
@@ -61,6 +61,74 @@
...
+    "regenerates": 10,
+    "extend": { "flags": [ "SLUDGETRAIL", "SLUDGEPROOF" ] }
+  },
+  {
+    "id": "mon_hulk_pupa",
+    "type": "MONSTER",
+    "name": { "str": "hive hulk" },
+    "description": "This gigantic corpse is wrapped in sticky black fibers that cover everything neck down.  Beneath the wrapping there is strange rhythmic movement grotesque to behold.",
+    "copy-from": "mon_hulk_pupa_decoy",
```
Comment (Maleclypse) [inline data/json/monsters/zed-pupating.json:120]:
```suggestion
    "copy-from": "mon_hulk_pupa_decoy",
```
I would suggest trying to break the circular json by not using copy-from here and adding the needed lines from where you are copying from.  If that doesn't work I'd look at whether you have created in the monster evolution's a loop where a monster evolves and copies from a monster that only exists via copy from.  That's probably a poor description but it's the best description I can think of for some tricky evolve messups that I've made previously.

Comment (SariusSkelrets) [reply data/json/monsters/zed-pupating.json]: I uncopied them and half the errors were gone. Thanks!

**Comment by Maleclypse** [comment]
Goto https://dev.narc.ro/cataclysm/format.html   copy a file that you've edited, then paste it into the website.  Hit the lint button.  If it doesn't tell you there is an error the copy that and paste it over the file you copied it out of.  Do this for each file.   If a file hits a syntax error when linting do the same thing at jsonlint.com and follow their instructions and once it says valid json run it through narc's linter.

**Comment by SariusSkelrets** [comment]
linted it all

**Comment by Maleclypse** [comment]
> linted it all
> 
> I'm now trying to fix infigthing problems as the blast make other zombies and raptors hostile to the hulk

I think the only way to fix that would be to eliminate the blast damage.  I do question why would there be blast damage if the hive critter doesn't die after spawning other zombies?

**Comment by SariusSkelrets** [comment]
blast damage in summon spell is the number of created monsters

setting it to zero tells the spell to create zero monsters

**Comment by Maleclypse** [comment]
Oops that's right.  I don't know why the blast would cause hostility then. I'll take a second look.

File: data/json/monsters/zed-pupating.json [hunk 5]
```diff
@@ -61,6 +61,145 @@
       ]
     }
   },
+  {
+    "id": "mon_brute_pupa_decoy",
+    "type": "MONSTER",
+    "name": { "str": "pupating brute" },
+    "description": "This muscular human corpse is wrapped in sticky black fibers that cover everything neck down.  Beneath the wrapping there is strange rhythmic movement grotesque to behold.",
```
Comment (actual-nh) [inline data/json/monsters/zed-pupating.json:68]:
```suggestion
    "description": "This muscular human corpse is wrapped in sticky black fibers that cover everything from the neck down.  Beneath the wrapping there is a strange rhythmic movement, grotesque to behold.",
```
How do the fibers look "sticky"?

File: data/json/monsters/zed-pupating.json [hunk 6]
```diff
@@ -61,6 +61,145 @@
...
+    "speed": 95,
+    "regenerates": 10,
+    "upgrades": { "half_life": 21, "into_group": "GROUP_BRUTE_PUPA" },
+    "extend": { "flags": [ "SLUDGETRAIL", "SLUDGEPROOF" ] }
+  },
+  {
+    "id": "mon_brute_pupa",
+    "type": "MONSTER",
+    "name": { "str": "pupating brute" },
+    "description": "This muscular human corpse is wrapped in sticky black fibers that cover everything neck down.  Beneath the wrapping there is strange rhythmic movement grotesque to behold.",
```
Comment (actual-nh) [inline data/json/monsters/zed-pupating.json:80]:
```suggestion
    "description": "This muscular human corpse is wrapped in sticky black fibers that cover everything from the neck down.  Beneath the wrapping there is a strange rhythmic movement, grotesque to behold.",
```

File: data/json/monsters/zed-pupating.json [hunk 7]
```diff
@@ -61,6 +61,145 @@
...
+        }
+      ]
+    }
+  },
+  {
+    "id": "mon_hulk_pupa_decoy",
+    "type": "MONSTER",
+    "name": { "str": "hive hulk" },
+    "//": "Not using copy_from as it messes with monster evolution.",
+    "description": "The bloated torso of this gigantic corpse is wrapped in sticky black fibers.  Beneath the wrapping there is strange rhythmic movement grotesque to behold.",
```
Comment (actual-nh) [inline data/json/monsters/zed-pupating.json:98]:
```suggestion
    "description": "The bloated torso of this gigantic corpse is wrapped in sticky black fibers.  Beneath the wrapping there is a strange rhythmic movement, grotesque to behold.",
```

File: data/json/monsters/zed-pupating.json [hunk 8]
```diff
@@ -61,6 +61,145 @@
...
+      "SLUDGEPROOF",
+      "FILTHY"
+    ]
+  },
+  {
+    "id": "mon_hulk_pupa",
+    "type": "MONSTER",
+    "name": { "str": "hive hulk" },
+    "//": "Not using copy_from as it messes with monster evolution.",
+    "description": "The bloated torso of this gigantic corpse is wrapped in sticky black fibers.  Beneath the wrapping there is strange rhythmic movement grotesque to behold.",
```
Comment (actual-nh) [inline data/json/monsters/zed-pupating.json:149]:
```suggestion
    "description": "The bloated torso of this gigantic corpse is wrapped in sticky black fibers.  Beneath the wrapping there is a strange rhythmic movement, grotesque to behold.",
```

File: data/json/monster_special_attacks/spells.json [hunk 9]
```diff
@@ -189,6 +189,22 @@
     "effect": "summon",
     "effect_str": "mon_spawn_raptor"
   },
+  {
+    "type": "SPELL",
+    "id": "hive_raptor_spawn",
+    "name": { "str": "Spawn Raptor" },
+    "description": "Summons a zombiespawn raptor without killing the zombie.",
```
Comment (actual-nh) [inline data/json/monster_special_attacks/spells.json:196]:
```suggestion
    "description": "Summons a zombie-spawn raptor without killing the zombie.",
```

**Review by actual-nh — COMMENTED** [review]
Interesting!

**Comment by I-am-Erk** [comment]
I think we need to make a lesser sludge trail before we add these.

**Pull request merged**